src/AsyncBridge/AsyncCompatLibExtensions.cs

```diff
@@ -85,5 +97,231 @@ public static ConfiguredTaskAwaitable ConfigureAwait(this Task task, bool contin
+    /// cref="T:System.Threading.CancellationTokenSource"/>.
+    /// </param>
+    /// <exception cref="T:System.ObjectDisposedException">The exception thrown when this <see
+    /// cref="T:System.Threading.CancellationTokenSource"/> has been disposed.
+    /// </exception>
+    /// <exception cref="T:System.ArgumentOutOfRangeException">
+    /// The exception thrown when <paramref name="delay"/> is less than -1 or 
+    /// greater than Int32.MaxValue.
+    /// </exception>
+    /// <remarks>
+    /// <para>
+    /// The countdown for the delay starts during this call.  When the delay expires, 
+    /// this <see cref="T:System.Threading.CancellationTokenSource"/> is canceled, if it has
+    /// not been canceled already.
+    /// </para>
+    /// <para>
+    /// Subsequent calls to CancelAfter will reset the delay for this  
+    /// <see cref="T:System.Threading.CancellationTokenSource"/>, if it has not been
+    /// canceled already.
+    /// </para>
+    /// </remarks>
+    public static void CancelAfter(this CancellationTokenSource cancelSource, TimeSpan delay)
+    {
+        if (cancelSource == null)
+            throw new ArgumentNullException(nameof(cancelSource));
+
+        if (delay < _timeoutInfinite)
+            throw new ArgumentOutOfRangeException(nameof(delay));
+
+        if (cancelSource.IsCancellationRequested)
+            return;
+
+        Timer MyTimer = null;
+
+        // CTS claims it's thread-safe for all methods, so we should be the same.
+        // If someone calls CancelAfter concurrently for the same CTS, we won't explode or cause a memory leak
+        do
+        {
+            if (_cancelTimers.TryGetTimer(cancelSource, out var CurrentTimer))
+            {
+                if (MyTimer != null)
+                    MyTimer.Dispose();
+
+                MyTimer = CurrentTimer;
+                break; // Reset the existing timer to the new delay
+            }
+
+            // An active timer doesn't hold a strong reference to itself, so adding the CTS as state won't hold it alive
+            MyTimer = new Timer(OnCancelAfterTimer, cancelSource, Timeout.Infinite, Timeout.Infinite);
+
+        } while (!_cancelTimers.TryAddTimer(cancelSource, MyTimer));
+
+        try
+        {
+            MyTimer.Change(delay, _timeoutInfinite);
+        }
+        catch (ObjectDisposedException)
+        {
+        }
+    }
+
+    private static void OnCancelAfterTimer(object state)
+    {
+        var CancelSource = (CancellationTokenSource)state;
+
+        if (!_cancelTimers.TryRemoveTimer(CancelSource, out var MyTimer))
+            return;
+
+        MyTimer.Dispose();
+
+        try
+        {
+            CancelSource.Cancel();
+        }
+        catch (ObjectDisposedException) // If the cancellation token has been disposed of, ignore the exception
+        {
+        }
+    }
+
+#if NET35
+    private struct WeakCTS : IEquatable<WeakCTS>
```

Comment:
`readonly struct`?

Reply:
Only available in C# 7.2. The current project is configured to use the latest major release, which is 7.0, unless you want me to change that...?

Reply:
Ah, right. I usually change it to `latest`. Don't worry about it (unless you want to).